### ISSUE TITLE
Add QuizKnock STADIUM (UKS) Support

### ISSUE THREAD

**Issue by smpn2** [spice2x/spice2x.github.io#131]
## Description of change
Support QuizKnock STADIUM Arcade.

## Testing

1. Get dump.
2. Create `prop\ea3-ident.xml`, or edit `prop\defaults\ea3-config.xml`.
```xml
<?xml version="1.0" encoding="SHIFT-JIS"?>
<ea3_conf>
  <soft>
    <model __type="str">UKS</model>
    <dest __type="str">J</dest>
    <spec __type="str">A</spec>
    <rev __type="str">A</rev>
    <ext __type="str">2000010100</ext>
  </soft>
</ea3_conf>
```
3. (Optional) If need change execexe temporary directory in `ee-config.xml`.
```sh
$ diff ee-config.xml ee-config.bak.xml
12c12
<   <tmp_dir __type="str">./dev/ee_tmp</tmp_dir>
---
>   <tmp_dir __type="str">c:\ee_tmp</tmp_dir>
```
4. Edit `prop\avs-config.xml`.
```sh
$ diff avs-config.xml avs-config.bak.xml
6d5
<       <option>posix=0</option>
13c12
<       <vfs name="boot" fstype="fs" src="./dev/e" dst="/dev/drive/e/" opt="vf=1,posix=1"/>
---
>       <vfs name="boot" fstype="fs" src="E:\" dst="/dev/drive/e/" opt="vf=1,posix=1"/>
54a54
>
```
5. Boot.
`spice64 -modules .\modules -url "http://[SERVER]" -urlslash 1 -p [PCBID] -e prop\defaults\ea3-config.xml -card0 [CARD ID]`

## Patch file
[QKS.txt](https://github.com/spice2x/spice2x.github.io/files/14099771/QKS.txt)

## Known Issues
 - No LED support.
 - Game does not call BI2X AddCounter function even if you press the coin switch. Therefore, coins will continue to be inserted as long as you press the button. (Inserting coins using the F1 key works without problems.)
 - Xonar AE does not output audio. Audio will be played on common Realtek and NVIDIA HDMI audio devices.

**Comment by smpn2** [comment]
@sp2xdev 
Yes. There is only one actual microswitch on the button, but it has an optical sensor that detects when it is lightly pressed (QButton1~3). 
As far as I know, this is only mentioned in the instruction operator manual and in the test menu, so I don't know if it's actually used.

Input Check
![image](https://github.com/spice2x/spice2x.github.io/assets/56753121/076dda74-3a33-4d3d-b329-27b98a3ca9d3)

**Comment by smpn2** [comment]
It seemed better to rename it, so I renamed it.
```patch
diff --git a/games/qks/io.cpp b/games/qks/io.cpp
index b9370bc..aad8da8 100644
--- a/games/qks/io.cpp
+++ b/games/qks/io.cpp
@@ -10,13 +10,13 @@ std::vector<Button> &games::qks::get_buttons() {
                 &buttons,
                 "Test",
                 "Service",
-                "Coin",
-                "QButton1",
-                "QButton2",
-                "QButton3",
-                "QButton",
-                "JackDetect",
-                "MicDetect"
+                "Coin Mech",
+                "Q Button Sensor1",
+                "Q Button Sensor2",
+                "Q Button Sensor3",
+                "Q Button",
+                "Headphone Detect",
+                "Microphone Detect"
         );
     }
 ```

**Comment by LupinThidr** [comment]
What is the button officially named? (i.e on eagate or during a tutorial)
I assumed Q Button was short for Quiz Button, but the manual refers to it as "早押しボタン" ("Quick Press Button"?) in diagrams and service menu. (and 早押しボタンセンサ1 2 3 for the senors)

**Comment by LupinThidr** [comment]
I tried to find what the sensor changes actually look like in-game. It seems like the game uses it to signal to other players that your hand is over the button.
https://youtu.be/Cd5Fp6iyfHY?si=cl7nrWzjuiydS4ir&t=178

I guess nobody would actually bind it... maybe someone trying to build a dedicated quiz knock controller

**Comment by smpn2** [comment]
For some reason, this game supports both Japanese and English only in test mode, so I've also attached a screenshot of the Japanese version.
@LupinThidr wrote, the Japanese name is "早押しボタンセンサ", "早押しボタン".

![image](https://github.com/spice2x/spice2x.github.io/assets/56753121/fd603e22-7174-4a4d-aceb-29dba9749370)
